add documentation on EESSI test suite (#110)
EESSI/docs: docs/test-suite/usage.md

# Using the EESSI test suite

This page covers the usage of the [EESSI test suite](https://github.com/EESSI/test-suite).

We assume you have already [installed and configured](installation-configuration.md) the EESSI test suite on your
system.

## Listing available tests

To list the tests that are available in the EESSI test suite,
use `reframe --list` (or `reframe -L` for short).

If you have properly [configured ReFrame](#Configuring-ReFrame), you should
see a (potentially long) list of checks in the output:

```
$ reframe --list
...
[List of matched checks]
- ...
Found 123 check(s)
```

!!! note

>     When using `--list`, checks are only generated based on available modules.

Reviewer: ```suggestion
    When using `--list`, checks are only generated based on modules that are available in the system where the `reframe` command is invoked.
```

  ```
  reframe --system example ...
  ```
* To let ReFrame only generate checks for the `gpu` partition of the system named `example`, use:
  ```
  reframe --system example:gpu ...
  ```

Use `--dry-run` to check the impact of using the `--system` option.

### Filtering by tags { #filter-tag }

To filter tests using one or more tags, you can use the [`--tag` option](https://reframe-hpc.readthedocs.io/en/stable/manpage.html#cmdoption-0).

Using `--list-tags` you can get a list of known tags.

To check the impact of this on generated checks by ReFrame, use `--list` or `--dry-run`.

#### `CI` tag

For each software that is included in the EESSI test suite,
a small test is tagged with `CI` to indicate it can be used in a Continuous Integration (CI) environment.

Hence, you can use this tag to let ReFrame only generate checks for small test cases:

```
reframe --tag CI
```

For example:

```
$ reframe --name GROMACS --tag CI
...
```

#### `scale` tags

The EESSI test suite defines a set of custom tags that control the *scale* of checks,
which specify many cores/nodes should be used for running a check.

| tag name | description |
|:--------:|-------------|

> | `1_core` | using a single CPU core, or single GPU |
> | `2_cores` | using 2 CPU cores, or 2 GPUs |
> | `4_cores` | using 4 CPU cores, or 4 GPUs |

Reviewer: this is not entirely correct
```suggestion
| `1_core` | using 1 CPU core and 1 GPU (if running a GPU test) |
| `2_cores` | using 2 CPU cores and 1 GPU (if running a GPU test) |
| `4_cores` | using 4 CPU cores, or 4 GPUs and 1 GPU (if running a GPU test) |
```

> | `4_cores` | using 4 CPU cores, or 4 GPUs and 1 GPU (if running a GPU test) |

Reviewer: ```suggestion
| `4_cores` | using 4 CPU cores and 1 GPU (if running a GPU test) |
```

## Overriding test parameters *(advanced)*

You can override test parameters using the [`--setvar` option (or `-S`)](https://reframe-hpc.readthedocs.io/en/stable/manpage.html#cmdoption-S).

This can be done either globally (for all tests), or only for specific tests (which is recommended when using `--setvar`).

For example, to run all GROMACS checks with a specific GROMACS module, you can use:

```
reframe --setvar GROMACS_EESSI.modules=GROMACS/2023.1-foss-2022a ...
```

!!! warning

    We do not recommend using `--setvar`, since it is quite easy to make unintended changes to test parameters
    this way that can result in broken checks.

    You should try filtering tests using the [`--name`](#filter-name) or [`--tag`](#filter-tag) options instead.

## Example commands

### Running all GROMACS tests on 4 cores on the `cpu` partition

```
reframe --run --system example:cpu --name GROMACS --tag 4_cores --performance-report
```

### List all checks for TensorFlow 2.11 using a single node

```
reframe --list --name %module_name=TensorFlow/2.11 --tag 1_node
```

> ### Dry run of TensorFlow CI checks on a quarter node (on all system partitions)

Reviewer: ```suggestion
### Dry run of TensorFlow CI checks on a quarter (1/4) of a node (on all system partitions)
```
